src/Traits/ErrorTrait.php

```diff
@@ -110,7 +111,7 @@ private static function assertValidName(string $name): void
             throw new InvalidArgumentException("A name cannot be an empty string");
         }
 
-        if (!ctype_alnum(str_replace('_', '', $name))) {
+        if (!preg_match('/^\p{L}[\p{L}0-9_]*$/u', $name)) {
             throw new InvalidArgumentException('A name can only contain alphanumeric characters and underscores');
         }
 
```

Comment:
The `is_numeric($name[0])` below can also be removed here, the regex checks for that